dll/win32/shell32/CDefView.cpp

```diff
@@ -1197,11 +1245,13 @@ BOOL CDefView::_Sort(int Col)
         }
 
         if (prevCol == Col)
-            m_sortInfo.bIsAscending = !m_sortInfo.bIsAscending;
+            m_sortInfo.Direction *= -1;
         else
-            m_sortInfo.bIsAscending = TRUE;
+            m_sortInfo.Direction = 0;
         m_sortInfo.ListColumn = Col;
     }
+    if (!m_sortInfo.Direction)
+        m_sortInfo.Direction += 1;
```

Comment:
So these two lines, just actually set `m_sortInfo.Direction` to 1 ? (since this is in the case `!m_sortInfo.Direction` being TRUE, i.e. `m_sortInfo.Direction == 0`).
If so, why not just doing `...Direction = 1;` instead of the += 1 ?

And further so, why not setting Direction = 1 in the else-case that is above (line 1250)?

Reply:
I can't set it on line 1250 because that path is not always hit. In general, the data in m_sortInfo is not fully trusted because several things outside DefView gets to play with the sorting.